### ISSUE TITLE
TransactionAbortedException when running Test while inserting Entities

### ISSUE THREAD

**Issue by brase** [zzzprojects/EntityFramework-Effort#85]
While using resharper in combination with nunit (single threaded) I am stumbled over the same Type of error mentioned in https://github.com/tamasflamich/effort/issues/48 with the same exception:

```
SetUp : System.Exception : Beim Herstellen der Verbindung zur Datenbank ist ein Fehler aufgetreten.
  ----> System.Data.UpdateException : Fehler beim Aktualisieren der Einträge. Weitere Informationen finden Sie in der internen Ausnahme.
  ----> System.Reflection.TargetInvocationException : Ein Aufrufziel hat einen Ausnahmefehler verursacht.
  ----> System.Transactions.TransactionAbortedException : Die Transaktion wurde abgebrochen.
   bei MediFoxRZ.Data.Mapping.ContextFactory.CreateContext() in D:\BFS\BfsFinance\DataAccess\MediFoxRZ.Data.Mapping\ContextFactory.cs:Zeile 28.
   bei MediFoxRZ.Services.Common.ServiceBase.Get[T](Func`2 func) in D:\BFS\BfsFinance\Services\MediFoxRZ.Services.Common\ServiceBase.cs:Zeile 15.
   bei MediFoxRZ.Services.Abrechnung.AbrechnungService.GetNachberechnungsgruende() in D:\BFS\BfsFinance\Services\MediFoxRZ.Services.Abrechnung\AbrechnungService.cs:Zeile 2571.
   bei MediFoxRZ.Services.Abrechnung.Tests.NachberechnungTests.Wenn_man_nach_existierenden_Nachberechnungsgruenden_fragt.Mit_Demo_Daten() in D:\BFS\BfsFinance\Services\MediFoxRZ.Services.Abrechnung.Tests\NachberechnungTests.cs:Zeile 30.
--UpdateException
   bei System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
   bei System.Data.Objects.ObjectContext.SaveChanges(SaveOptions options)
   bei MediFoxRZ.Services.Tests.With_InMemory_Db.<With_initialized_Entities>b__3_0(String constring) in D:\BFS\BfsFinance\Services\MediFoxRZ.Services.Tests\With_InMemory_Db.cs:Zeile 40.
   bei MediFoxRZ.Data.Mapping.ContextFactory.CreateContext() in D:\BFS\BfsFinance\DataAccess\MediFoxRZ.Data.Mapping\ContextFactory.cs:Zeile 24.
--TargetInvocationException
   bei System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   bei System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   bei System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   bei System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   bei Effort.Internal.Common.DatabaseReflectionHelper.InsertEntity(ITable table, Object entity, Transaction transaction) in C:\Users\brase\code\rs\spielwiese\effort\Main\Source\Effort\Internal\Common\DatabaseReflectionHelper.cs:Zeile 125.
   bei Effort.Internal.CommandActions.InsertCommandAction.CreateAndInsertEntity(ITable table, IList`1 memberBindings, Transaction transaction) in C:\Users\brase\code\rs\spielwiese\effort\Main\Source\Effort\Internal\CommandActions\InsertCommandAction.cs:Zeile 159.
   bei Effort.Internal.CommandActions.InsertCommandAction.ExecuteNonQuery(ActionContext context) in C:\Users\brase\code\rs\spielwiese\effort\Main\Source\Effort\Internal\CommandActions\InsertCommandAction.cs:Zeile 137.
   bei Effort.Provider.EffortEntityCommand.ExecuteNonQuery() in C:\Users\brase\code\rs\spielwiese\effort\Main\Source\Effort\Provider\EffortEntityCommand.cs:Zeile 92.
   bei System.Data.Mapping.Update.Internal.DynamicUpdateCommand.Execute(UpdateTranslator translator, EntityConnection connection, Dictionary`2 identifierValues, List`1 generatedValues)
   bei System.Data.Mapping.Update.Internal.UpdateTranslator.Update(IEntityStateManager stateManager, IEntityAdapter adapter)
--TransactionAbortedException
   bei NMemory.Transactions.Transaction.EnsureTransaction(Transaction& transaction, IDatabase database)
   bei NMemory.Tables.Table`2.Insert(TEntity entity, Transaction transaction)
   bei Effort.Internal.Common.DatabaseReflectionHelper.WrapperMethods.InsertEntity[TEntity](ITable`1 table, TEntity entity, Transaction transaction) in C:\Users\brase\code\rs\spielwiese\effort\Main\Source\Effort\Internal\Common\DatabaseReflectionHelper.cs:Zeile 285.
```
This happens only on some machines. Is there somewhere hidden state (filesystem?) in effort or nmemory which can cause such errors?

**Comment by JonathanMagnan** [comment]
Hello @brase 

Is it possible for you to create a sample project with this issue? My developer was not able to get this error.

We are currently making an issue cleanup and make sure we focus on issues people still require since it will be impossible to make them all (It's a free library!). Since that's has been a long time that this issue has been answered, it might also be already fixed.

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello,

This issue will be closed since none provided information.

We understand this issue has been created long ago, but since that's a free library, we want to make sure to work on an issue that is still active since we cannot make them all.

Feel free to reopen if you can provide further information.

Best Regards,

Jonathan

**Comment by mgolubovic82** [comment]
I'm reporting my findings here in case anyone runs into this and has the same problem I did.  

In my case it had to do with the save transaction timing out in some cases, but not others.  The exception was not clear that this was happening, it was just a transaction aborted exception as the inner most exception.   I was able to get around it by setting the Effort connection to a large number which is available in 2.0.5 I believe.   

 var connection = Effort.DbConnectionFactory.CreateTransient();
            connection.SetConnectionTimeout(500000);
